lib/telemetry_metrics_statsd/udp.ex

```diff
@@ -18,43 +16,53 @@ defmodule TelemetryMetricsStatsd.UDP do
   @spec open(config()) ::
           {:ok, t()} | {:error, reason :: term()}
   def open(config) do
-    opts = [{:active, false}]
-
-    opts =
-      Enum.reduce(config, opts, fn
-        {:host, {:local, _}}, opts -> [:local | opts]
-        {:inet_address_family, value}, opts -> [value | opts]
-        {_key, _value}, opts -> opts
-      end)
-
-    case :gen_udp.open(0, opts) do
-      {:ok, socket} ->
-        udp = struct(__MODULE__, Map.put(config, :socket, socket))
-        {:ok, udp}
-
-      {:error, _} = err ->
-        err
+    {domain, address} =
+      case config.host do
+        {:local, path} ->
+          {:local, %{family: :local, path: path}}
+
+        ip when tuple_size(ip) == 4 ->
+          {:inet, %{family: :inet, port: config.port, addr: ip}}
+
+        ip when tuple_size(ip) == 8 ->
+          {:inet, %{family: :inet6, port: config.port, addr: ip}}
+      end
+
+    with {:ok, socket} <- :socket.open(domain, :dgram),
+         :ok <- :socket.connect(socket, address) do
+      udp = struct(__MODULE__, Map.put(config, :socket, socket))
+      {:ok, udp}
     end
   end
 
   @spec send(t(), iodata()) :: :ok | {:error, reason :: term()}
-  def send(%__MODULE__{host: host, port: port, socket: socket}, data) do
-    case host do
-      {:local, _} ->
-        :gen_udp.send(socket, host, 0, data)
-
-      _ ->
-        :gen_udp.send(socket, host, port, data)
-    end
+  def send(%__MODULE__{socket: socket}, data) do
+    :socket.send(socket, data)
+    |> handle_send_result()
   end
 
   @spec update(t(), :inet.hostname() | :inet.ip_address(), :inet.port_number()) :: t()
-  def update(%__MODULE__{} = udp, new_host, new_port) do
-    %__MODULE__{udp | host: new_host, port: new_port}
+  def update(%__MODULE__{} = udp, _new_host, _new_port) do
+    #%__MODULE__{udp | host: new_host, port: new_port}
```

Comment:
Figure out how to do updates for a connected socket. I suppose the only option is to reconnect?